### ISSUE TITLE
[GUIDE] how to not enable vkbasalt by default

### ISSUE THREAD

**Issue by VoodaGod** [simons-public/steam-deck-vkbasalt-install#2]
if you only want to use vkbasalt for a few select games, instead of setting `DISABLE_VKBASALT=1` for every single other game individually, remove these lines from the script before running it:
https://github.com/simons-public/steam-deck-vkbasalt-install/blob/1af3ac27e48e0de7f24d924217e9454a4efac559/vkbasalt_install.sh#L45 and https://github.com/simons-public/steam-deck-vkbasalt-install/blob/1af3ac27e48e0de7f24d924217e9454a4efac559/vkbasalt_install.sh#L50
or if already installed change 
```
    "enable_environment": {
      "SteamDeck": "1"
    },
```
to 
```
    "enable_environment": {
      "ENABLE_VKBASALT": "1"
    },
```
in `~/.local/share/vulkan/implicit_layer.d/vkBasalt.json` and `~/.local/share/vulkan/implicit_layer.d/vkBasalt.x86.json`

now you can enable vkbasalt for individual games with `ENABLE_VKBASALT=1` as usual.

one thing to note is that vkbasalt will not be applied to gamescope, but to the game itself, which means it won't work for games not using vulkan (directly or through eg. dxvk/vkd3d).
for opengl games, you can simply use the script described here https://github.com/DadSchoorse/vkBasalt/issues/168#issuecomment-1675866422 to force the game to run with zink and apply vkbasalt to that

**Comment by MrPRAINGLE** [comment]
Hey, any idea on how to make it work on desktop ? This may work for the Steam Deck but on my PC, the file (located at /usr/share/vulkan/implicit_later.d/vkBasalt.json) is already modified to be that way, and vkBasalt seems to be active on most applications I launch (Snapchat, FreeTube...)

**Comment by VoodaGod** [comment]
@MrPRAINGLE is `ENABLE_VKBASALT` set globally for you somehow? try `echo $ENABLE_VKBASALT`, normally this should be empty, and you would set `ENABLE_VKBASALT=1` when you launch an application you want to use vkbasalt.
how is vkbasalt installed for you?

**Comment by MrPRAINGLE** [comment]
The variable is set to 1, I edited .bashrc adding "export ENABLE_VKBASALT=0" and "export DISABLE_VKBASALT=1". In the terminal the variables are changed, but when launching ann app from the app launcher, vkBasalt seems to still be enabled. And no, I did not enable it in the .desktop file. Any idea why ?

**Comment by MrPRAINGLE** [comment]
Oh and I did not mention it, but I cloned and installed vkBasalt from the repo with the following commands:
```
git clone https://github.com/DadSchoorse/vkBasalt.git
cd vkBasalt
meson setup --buildtype=release --prefix=/usr builddir
ninja -C builddir install
```

**Comment by VoodaGod** [comment]
ok i thought you might have installed it as a distribution package which sets the environment variable

i see 2 options:
- check the files described in https://wiki.archlinux.org/title/environment_variables to find out where the `ENABLE_VKBASALT=1` is coming from & get rid of it
- adjust your `vkBasalt.json` to use a different environment variable to enable the vkbasalt layer, eg:
```
    "enable_environment": {
      "VKBASALT_ENABLE": "1"
    },
```
and then selectively enable it like eg. `VKBASALT_ENABLE=1 %command%` for a steam game

**Comment by MrPRAINGLE** [comment]
Hey ! After looking up in the files listed in the wiki page you provided, I could not find where the variables were declared. I typed `export | grep VK` and it still showed that ENABLE_VKBASALT was set to 0 and DISABLE_VKBASALT was set to 1, so the variables must have been declared somewhere else.

I ended up uninstalling the APT package completely (which I assume was causing the issue) and reinstalling vkBasalt entirely, and now it is not enabled by default, and `export | grep VK` does not return anything, so, problem solved ! Thank you for your help !

**Comment by Ex0ZX** [comment]
How do you install this? Every time I put in bash vkbasalt_install.sh and press enter, the terminal pauses for a moment and then nothing happens. Nothing is installed. Please help.

**Comment by simons-public** [comment]
Can you run it with 'bash -x vkbasalt_install.sh' and post the output?

**Comment by Ex0ZX** [comment]
(1)(deck@steamdeck ~)$ bash -x vkbasalt_install.sh
+ set -eo pipefail
+ grep -q SteamOS /etc/os-release
+ '[' 1000 -eq 0 ']' 
+ DECK_ HOME=/home/deck
+ AUR_BASE=https://builds.garudalinux.org/repos/chaotic-aur/x86_64/
++ curl https://builds.garudalinux.org/repos/chaotic-aur/x86_64/
++ grep -o 'vkbasalt-[0-9\.]*-1-x86_64'
++ head -1
+ VKBASALT_PKG_VER-
(1)(deck@steamdeck ~)$

**Comment by simons-public** [comment]
The package regex wasn't matching anymore. I'm on mobile so I can't verify, but it should be fixed now if you try the last update

**Comment by Ex0ZX** [comment]
Thanks for the update but now I got a 404 error;

2025-03-14 19:30:12 (763 KB/s) - ‘vkbasalt_install.sh’ saved [2727/2727]

(deck@steamdeck ~)$ bash vkbasalt_install.sh
--2025-03-14 19:30:46--  https://builds.garudalinux.org/repos/chaotic-aur/x86_64/vkbasalt-0.3.2.10-1.1-x86_64.pkg.tar.zst%22%3Elib32-vkbasalt-0.3.2.10-1.1-x86_64.pkg.tar.zst
Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving builds.garudalinux.org (builds.garudalinux.org)... 116.202.208.112
Connecting to builds.garudalinux.org (builds.garudalinux.org)|116.202.208.112|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://cf-builds.garudalinux.org/repos/chaotic-aur/x86_64/vkbasalt-0.3.2.10-1.1-x86_64.pkg.tar.zst%22%3Elib32-vkbasalt-0.3.2.10-1.1-x86_64.pkg.tar.zst [following]
--2025-03-14 19:30:47--  https://cf-builds.garudalinux.org/repos/chaotic-aur/x86_64/vkbasalt-0.3.2.10-1.1-x86_64.pkg.tar.zst%22%3Elib32-vkbasalt-0.3.2.10-1.1-x86_64.pkg.tar.zst
Resolving cf-builds.garudalinux.org (cf-builds.garudalinux.org)... 172.67.75.71, 104.26.15.216, 104.26.14.216, ...
Connecting to cf-builds.garudalinux.org (cf-builds.garudalinux.org)|172.67.75.71|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2025-03-14 19:30:47 ERROR 404: Not Found.

(8)(deck@steamdeck ~)$

**Comment by simons-public** [comment]
Yeah regex was too greedy, just pushed an update, should be fixed now. Thanks for testing

**Comment by Ex0ZX** [comment]
Thanks for the quick update. It finally installed. Have a good one 👍
